server/app/services/PhoneValidationUtils.java

```diff
@@ -7,45 +7,110 @@
 
 /** Utility Class for all phone validations. */
 public final class PhoneValidationUtils {
+  private static final String[] POTENTIAL_COUNTRY_CODES = new String[] {"US", "CA"};
 
   private static final PhoneNumberUtil PHONE_NUMBER_UTIL = PhoneNumberUtil.getInstance();
 
+  // Just a small container object to return multiple values *within* this class. Can be
+  // replaced with Tuple when we get to Java 17+. If we need to expose this outside of
+  // the class convert it to be an @AutoValue-based class.
+  private static final class PhoneValidationResult {
+    public Optional<String> countryCode;
+    public Optional<MessageKey> messageKey;
+
+    private PhoneValidationResult(Optional<String> countryCode, Optional<MessageKey> messageKey) {
+      this.countryCode = countryCode;
+      this.messageKey = messageKey;
+    }
+
+    public static PhoneValidationResult create(String countryCode) {
+      return new PhoneValidationResult(Optional.of(countryCode), Optional.empty());
+    }
+
+    public static PhoneValidationResult createError(MessageKey messageKey) {
+      return new PhoneValidationResult(Optional.empty(), Optional.of(messageKey));
+    }
+  }
+
   /**
-   * Gets an Optional string of Phone number and CountryCode values and runs validations on them.
+   * Gets an Optional MessageKey of Phone number and CountryCode values and runs validations on
+   * them.
    *
-   * @return an Optional {@link MessageKey} representing the validation error that occurred if the
+   * @param phoneNumber {@link Optional<String>}} of the Phone Number
+   * @param countryCode {@link Optional<String>}} of the Country Code
+   * @return an {@link Optional<MessageKey>} representing the validation error that occurred if the
    *     phone number is invalid, or an Optional.empty() if the phone number is valid
    */
   public static Optional<MessageKey> validatePhoneNumber(
       Optional<String> phoneNumber, Optional<String> countryCode) {
-    if (phoneNumber.isEmpty()) {
-      return Optional.of(MessageKey.PHONE_VALIDATION_NUMBER_REQUIRED);
+    var result = validate(phoneNumber, countryCode);
+    return result.messageKey;
+  }
+
+  /**
+   * Gets an Optional string of Phone number and CountryCode values and runs validations on them.
+   *
+   * @param phoneNumber {@link Optional<String>}} of the Phone Number
+   * @return an {@link Optional<String>} representing the country code if the phone number is valid,
+   *     or an Optional.empty() if the phone number is invalid
+   */
+  public static Optional<String> determineCountryCode(Optional<String> phoneNumber) {
```

Comment:
**suggestion (non-blocking)**: not-blocking cause it might come up as you do the rest of this refactor, but I'd prefer we return a more clear type, rather than assigning meaning to the empty optional.

Also we're assigning the opposite meaning to the empty optional than we do in `validatePhoneNumber()`, which is likely going to be confusing to reason about at the callsite.

Options for a return type could include adding a `phoneNumber` field to `PhoneValidationResult` and passing that out of the class, or using `ErrorAnd` to wrap the error and the result:
https://github.com/civiform/civiform/blob/b10353fdb25c81f2c65e29935b8c96bcefa02839/server/app/services/ErrorAnd.java#L13-L13
We do both of these things elsewhere so I think either is fine.

Reply:
I'm expecting to remove the existing `validatePhoneNumber` that only returns an `Optional<MessageKey>` leaving just one that will return the `PhoneValidationResult`.